### ISSUE TITLE
Data Loss

### ISSUE THREAD

**Issue by Maxi605** [GuskiS/Trouble-in-Terrorist-Town#22]
Sometimes when a player DC's or gets kicked by the server (Maybe banned) his data gets deleted when he rejoins, he has 1.000.000 points (Example) and when he gets in again, 0.

**Comment by GuskiS** [comment]
You mean karma saving thing? It is hard to understand this one.

**Comment by Maxi605** [comment]
No, i'm talking about the data that saves the server like your points, kills as T, kills as D, kills as I, etc.

**Comment by GuskiS** [comment]
They are called stats. Anyways, will check, but needs a bit more info.

**Comment by Maxi605** [comment]
Not much info on it, when a client DC's sometimes it looses the stats, when i used a anti-retry, the player gets kicked for doing retry and then is when the stats gets lost, he starts from zero, we fixed that and it worked but it seems to be another trigger for it or something

**Comment by GuskiS** [comment]
If you could do some sort of testing that could help.
https://github.com/GuskiS/Trouble-in-Terrorist-Town/blob/1ddd5856b8b7b8ea15190ff1bb0f5d5d4ba57996/scripting/ttt_stats_system.sma#L199
before this line, add `log_amx("TTT - MySQL_Load: %d  == %s", id, g_szPlayerName[id])`

=======================================

https://github.com/GuskiS/Trouble-in-Terrorist-Town/blob/1ddd5856b8b7b8ea15190ff1bb0f5d5d4ba57996/scripting/ttt_stats_system.sma#L230
before this line, add `log_amx("TTT - MySQL_LoadData: %d == %s == %d", id, g_szPlayerName[id], SQL_NumResults(query))`

=======================================

https://github.com/GuskiS/Trouble-in-Terrorist-Town/blob/1ddd5856b8b7b8ea15190ff1bb0f5d5d4ba57996/scripting/ttt_stats_system.sma#L379
before this line, add `log_amx("TTT - table_insert: %d  == %s", id, g_szPlayerName[id], name)` where name is `get_user_name` functions result.

**Comment by GuskiS** [comment]
Let me know if those results when something weird happens.
Also, do check if there are no duplicates in database for that `player_name`

**Comment by Maxi605** [comment]
Ok, i'll give upload it to the server.

**Comment by Maxi605** [comment]
The problem doesn't just happen with 1 name, it happens randomly (there's a trigger obviously) when the client attemps a conection to the server
